tests/Backend/SOX/MergeRequestsTest.php

```diff
@@ -93,6 +93,67 @@ public function testPutInReview(): void
         $this->assertEquals('in_review', $mrData['state']);
     }
 
+    public function testMRWorkflowFromDevelopmentToCancel(): void
+    {
+        $oldBranches = $this->branches->listBranches();
+        $this->assertCount(1, $oldBranches);
+
+        $newBranch = $this->branches->createBranch('aaaa');
+
+        $mrId = $this->developerClient->createMergeRequest([
+            'branchFromId' => $newBranch['id'],
+            'branchIntoId' => $oldBranches[0]['id'],
+            'title' => 'Change everything',
+            'description' => 'Fix typo',
+        ]);
+
+        $reviewerClient = $this->getReviewerStorageApiClient();
+        $this->developerClient->mergeRequestPutToReview($mrId);
+
+        $mrData = $reviewerClient->mergeRequestAddApproval($mrId);
+
+        $this->assertEquals('in_review', $mrData['state']);
+        $this->assertCount(1, $mrData['approvals']);
+
+        $mrData = $reviewerClient->rejectMergeRequest($mrId);
+        $this->assertCount(0, $mrData['approvals']);
+        $this->assertSame('development', $mrData['state']);
+
+        $mrData = $reviewerClient->cancelMergeRequest($mrId);
+        $this->assertCount(0, $mrData['approvals']);
+        $this->assertSame('canceled', $mrData['state']);
+        $this->assertSame(null, $mrData['branches']['branchFromId']);
+    }
+
+    public function testAddSingleApprovalOnly(): void
+    {
+        $oldBranches = $this->branches->listBranches();
+        $this->assertCount(1, $oldBranches);
+
+        $newBranch = $this->branches->createBranch('aaaa');
+
+        $mrId = $this->developerClient->createMergeRequest([
+            'branchFromId' => $newBranch['id'],
+            'branchIntoId' => $oldBranches[0]['id'],
+            'title' => 'Change everything',
+            'description' => 'Fix typo',
+        ]);
+
+        $reviewerClient = $this->getReviewerStorageApiClient();
+        $this->developerClient->mergeRequestPutToReview($mrId);
+
+        $mrData = $reviewerClient->mergeRequestAddApproval($mrId);
+
+        $this->assertEquals('in_review', $mrData['state']);
+        $this->assertCount(1, $mrData['approvals']);
+
+        try {
+            $mrData = $reviewerClient->mergeRequestAddApproval($mrId);
+        } catch (ClientException $e) {
+            $this->assertSame('Operation canot be performed due: This reviewer has already approved this request.', $e->getMessage());
+        }
+    }
```

Comment:
Este by som otestoval, ze ked pridam druhy approve tak sa to naozaj approvne

Reply:
jo, na to ale musim predelat generator projektu, protoze jak jsi psal v minulem komentari - je na to potreba dalsi token s REVIEWER roli. udelam si na to dalsi task

Reply:
Ok alebo myslis ze by to slo napr zablokovat az v momente toho ked budeme mat dorieseny ten test ? Pretoze vlastne bez toho nemozem napisat testy na svoje mergovanie, pretoze si to nemam ako approvnut :) Alebo predpokladas ze to bude nieco rychle :)